### PR TITLE
fix(openai_responses): make input_items pagination usable for ID-less input arrays

### LINKED ISSUE #548: Input-item pagination becomes unusable for ordinary ID-less input arrays
## Clawpatch finding

- **Severity:** medium
- **Category:** api-contract
- **Confidence:** high
- **Triage:** contract-mismatch
- **Feature:** `feat_custom_openai_responses_state_store`
- **Finding ID:** `fnd_sig-feat-custom-openai-responses_7e0f01710b`
- **Run ID:** `20260727T064626-81bf93`
- **Prior-run overlap:**
No prior-run overlap detected.

## Evidence

- `apis/src/openai/responses/store/input_items.rs:132-147` (`normalize_input_items`) — Array input is returned verbatim, without assigning resource IDs.
- `apis/src/openai/responses/store/input_items.rs:121-128` (`list_input_items`) — An internal numeric next_cursor is calculated for pages whose items lack IDs.
- `apis/src/openai/responses/store/filter.rs:929-940` (`build_input_items_ok`) — The serialized response discards next_cursor and derives last_id only from the returned item.
- `apis/src/openai/responses/store/tests.rs:3654-3671` (`list_input_items_next_cursor_falls_back_to_numeric_offset`) — The test validates the internal numeric fallback but not its omission from the HTTP response.

## Description

Responses create requests commonly contain input item parameters without server resource IDs. When such input spans multiple pages, the endpoint returns has_more=true with last_id=null and omits the internally generated numeric cursor, leaving the client no value it can pass as after to retrieve the next page; it also returns parameter-shaped objects rather than consistently normalized item resources.

## Reproduction

Persist an input array containing two message objects without id fields. Request /v1/responses/{id}/input_items?limit=1 and observe a response with one item, has_more=true, last_id=null, and no usable continuation cursor.

## Recommendation

Assign stable resource IDs to all stored input items, preferably at persistence time, and paginate exclusively with those IDs. Ensure first_id and last_id are populated whenever data is nonempty and has_more can be followed using the documented after parameter.

## Notes

Generated by Clawpatch review over `apis/src` excluding `apis/src/token_usage/**` after rebasing on `origin/main`. Please validate before assigning or starting implementation.

### PR DESCRIPTION
**Pull request by mkoushni**
## Summary

- `GET /v1/responses/{id}/input_items` paginates ID-less array `input` items using an internal numeric offset, but `first_id`/`last_id` in the response only ever look at each item's `id` field. Plain content-part objects in the `input` array carry no `id`, so `last_id` stays `null` even when `has_more: true` — clients have no `after` value to resume pagination, even though the resolver already computed a usable numeric cursor internally.
- Fix: assign a stable synthetic `id` (`msg_{response_id}_input_{index}`) to any input item missing one, in `normalize_input_items`. The ID is keyed to the item's original stored position, so it stays stable regardless of `order` (ascending/descending). IDs become the primary pagination cursor, matching the existing `scalar_input_item` pattern already used for scalar `input`.
- As defense-in-depth, `build_input_items_ok` now also falls back to the page's numeric `next_cursor` for `last_id` when the last item genuinely has no ID (e.g. a non-object array entry like a plain string, which can't carry a synthetic ID). This closes the one remaining gap and keeps `after`-based pagination usable in every case, not just the common one.
- No changes were needed to how `first_id`/`last_id` are extracted from `page.data` — that logic was already correct; it simply had no IDs to read before.

This is the smallest complete change: it only touches the read-path normalization and response-building functions plus their surrounding tests, and does not touch persistence, schema, or the create/store path.

## Related issue

Closes #548

## Validation

```console
cargo test -p praxis-ai-apis --lib
# test result: ok. 2014 passed; 0 failed; 25 ignored

cargo clippy -p praxis-ai-apis --all-targets
# Finished with 0 warnings

cargo +nightly fmt --check -p praxis-ai-apis
# no output (clean)
```

New/updated tests cover:

- `list_input_items_assigns_synthetic_ids_to_id_less_array_items` — synthetic IDs stay tied to original position across sort order.
- `list_input_items_next_cursor_uses_synthetic_id_when_input_lacks_ids` / `list_input_items_numeric_cursor_still_works_as_fallback_for_synthetic_ids` — updated/added to reflect ID-based cursors, with numeric cursors still accepted for backward compatibility.
- `get_input_items_pagination_usable_for_id_less_array_items` — end-to-end HTTP regression reproducing the original bug: two ID-less items, `limit=1`, confirms `last_id` is non-null and that `after=<last_id>` correctly resumes to the second item.
- `get_input_items_last_id_falls_back_to_numeric_cursor_for_non_object_items` — covers the residual non-object array entry edge case.
- `store::tests::input_items_from_array_input` / `input_items_limit_zero_clamps_to_one` — updated to expect synthetic-ID cursors instead of raw numeric ones.

## Breaking changes

None. `first_id`/`last_id` were previously `null` for ID-less arrays; they are now populated. Any client already tolerating `null` (per the documented contract) is unaffected; clients blocked by the bug can now paginate correctly.

### CODE REVIEW

File: apis/src/openai/responses/store/input_items.rs [hunk 1]
```diff
@@ -137,14 +137,43 @@ pub fn list_input_items(record: &ResponseRecord, params: &ListParams) -> Result<
...
+/// Assign a stable synthetic ID (`msg_{response_id}_input_{index}`) to
+/// any item missing one, keyed by its position in the original stored
+/// input order (before any order-based reversal).
+///
+/// Plain content-part objects — the common shape for array `input` —
+/// carry no `id` field. Without a synthetic one, `first_id`/`last_id`
+/// in the list response stay `null` and clients have no `after` value
+/// to resume pagination past the first page, even though `has_more`
+/// reports `true`.
+fn ensure_stable_ids(response_id: &str, items: Vec<serde_json::Value>) -> Vec<serde_json::Value> {
```
[the remark this answers is not in the document]
Comment (aslakknutsen) [reply apis/src/openai/responses/store/input_items.rs:162]: @mkoushni any comment to this? ^ beyond that it looks good :)

[the remark this answers is not in the document]
Comment (mkoushni) [reply apis/src/openai/responses/store/input_items.rs:162]: i'm in vacation.fixed it now

**Review by leseb — APPROVED** [review]
